sale_import_base/tests/common_sale_order_import.py

```diff
@@ -0,0 +1,202 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from copy import deepcopy
+
+from odoo.addons.component.tests.common import SavepointComponentCase
+from odoo.addons.datamodel.tests.common import SavepointDatamodelCase
+from odoo.addons.sale.tests.test_sale_common import TestCommonSaleNoChart
+
+
+class SaleImportCase(
+    SavepointDatamodelCase, TestCommonSaleNoChart, SavepointComponentCase
+):
+    @classmethod
+    def setUpClass(cls):
+        super(SaleImportCase, cls).setUpClass()
+        cls.setUpClassicProducts()
+        cls.setUpAddresses()
+        cls.setUpLines()
+        cls.setUpImport()
+        cls.setUpTaxes()
+        cls.setUpFpos()
+        cls.setUpPayment()
+        cls.setUpMisc()
+
+    @classmethod
+    def setUpAddresses(cls):
+        cls.addr_customer_example = {
+            "name": "Thomas Jean",
+            "street": "1 rue de Jean",
+            "street2": "bis",
+            "zip": "69100",
+            "city": "Lyon",
+            "email": "thomasjean@gmail.com",
+            "country_code": "FR",
+            "external_id": "ThomasJeanEbay",
+        }
+        cls.addr_shipping_example = {
+            "name": "shipping contact name",
+            "street": "2 rue de shipping",
+            "street2": "bis",
+            "zip": "69100",
+            "city": "Lyon",
+            "email": "not-required@not-required.com",
+            "country_code": "FR",
+        }
+        cls.addr_invoicing_example = {
+            "name": "invoicing contact name",
+            "street": "3 rue de invoicing",
+            "street2": "bis",
+            "zip": "69100",
+            "city": "Lyon",
+            "email": "whatever@whatever.io",
+            "country_code": "FR",
+        }
+
+    @classmethod
+    def setUpLines(cls):
+        cls.line_valid_1 = {
+            "product_code": "PROD_ORDER",
+            "qty": 5,
+            "price_unit": 1111.1,
+            "description": "Initial Line 1 import description",
+            "discount": 10.0,
+        }
+        cls.line_valid_2 = {
+            "product_code": "PROD_DEL",
+            "qty": 2,
+            "price_unit": 2222.2,
+            # description: missing
+            "discount": 0.0,
+        }
+        cls.line_invalid = {
+            "product_code": "DOES NOT EXIST",
+            "qty": "should not happen",
+            "price_unit": "wrong input",
+            "description": "",
+            "discount": 0.0,
+        }
+
+    @classmethod
+    def setUpImport(cls):
+        amt_untaxed_1 = (
```

Comment:
avoid computation of data it's hard to read, we do not dynamique stuff here.
A simple json with the data is enought